### PR TITLE
NF: remove custom json

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
it was only required to avoid checked exception. Thanks to Kotlin, it's not a concern anymore!

It's morally a revert of #5550 , reviewed by @mikehardy. 

But, even if I didn't actually test it, I'm quite certain that just `revert 4d3189284101f7aeaaebd32fa890812484aefde5` would not have worked

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/utils/JSONArray.kt [hunk 1]
```diff
@@ -41,388 +41,136 @@
...
-                            x.back()
-                        }
-                        ']' -> return
-                        else -> throw x.syntaxError("Expected a ',' or ']'")
-                    }
-                }
+import org.json.JSONArray
+import org.json.JSONObject
+
+@KotlinCleanup("simplify with when")
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/utils/JSONArray.kt:47]: cleanup can be removed, as it was solved

**Review by BrayanDSO — APPROVED** [review]
LGTM. Nothing worth blocking

File: AnkiDroid/src/main/java/com/ichi2/utils/JSONArray.kt (hunk 1 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/utils/JSONArray.kt]: Done.
I also splitted the commit in multiple simpler commit. Thanks for reviewing this as a whole, I admit myself, on a new day, I had trouble

**Comment by Arthur-Milchior** [comment]
Conflict resolved

**Review by mikehardy — APPROVED** [review]
LGTM - nice net code drop, and I like *not* having a custom JSON implementation in a flashcard app. Seems that's something "the world" should provide as a basic service ;-)

I pushed a fix for the conflict

Running these 4 commits through `./tools/test-commits.sh` but the last one already passed at least (I verified that before pushing my conflict resolution)

**Comment by mikehardy** [comment]
This passed `./tools/test-commits.sh` - good to go

**Pull request merged**